hierarchicalforecast/core.py

```diff
@@ -119,13 +122,55 @@ def _prepare_fit(
         id_col: str = "unique_id",
         time_col: str = "ds",
         target_col: str = "y",
-    ) -> tuple[FrameT, FrameT, FrameT, list[str]]:
+        id_time_col: str = "temporal_id",
+        temporal: bool = False,
+    ) -> tuple[FrameT, FrameT, FrameT, list[str], str]:
         """
         Performs preliminary wrangling and protections
         """
         Y_hat_nw_cols = Y_hat_nw.columns
         S_nw_cols = S_nw.columns
 
+        # Check if Y_hat_df has the necessary columns for temporal
+        if temporal:
+            # We don't support insample methods, so Y_df must be None
+            if Y_nw is not None:
+                raise NotImplementedError(
+                    "Temporal reconciliation requires `Y_df` to be None."
+                )
+            # If Y_nw is None, we need to check if the reconcilers are not insample methods
+            for reconciler in self.orig_reconcilers:
+                if reconciler.insample:
+                    reconciler_name = _build_fn_name(reconciler)
+                    raise NotImplementedError(
+                        f"Temporal reconciliation is not supported for `{reconciler_name}`."
+                    )
+            # Hence we also don't support bootstrap or permbu (rely on insample values)
+            if intervals_method in ["bootstrap", "permbu"]:
+                raise NotImplementedError(
+                    f"Temporal reconciliation is not supported for intervals_method=`{intervals_method}`."
+                )
+
+            missing_cols_temporal = set(
+                [id_col, time_col, target_col, id_time_col]
+            ) - set(Y_hat_nw_cols)
+            if len(missing_cols_temporal) > 0:
```

Comment:
the pythonic way of checking if a collection is not empty is checking is truthiness, so this should be `if missing_cols_temporal:`